### ISSUE TITLE
N804 false positive for `class Bar(type(foo))`

### ISSUE THREAD

**Issue by alexhenman** [astral-sh/ruff#14675]
```python
foo = {}

class Bar(type(foo)):
    def foo_method(self):
        pass
```

https://play.ruff.rs/240f9b93-2408-4aeb-814d-bb09629496da

Not sure if this might be related to the issue reported in #12736 and its fix

**Comment by AlexWaygood** [comment]
> Not sure if this might be related to the issue reported in [#12736](https://github.com/astral-sh/ruff/issues/12736) and its fix

yes, it is.

Hmm, not sure what to do here. This is hard to fix properly without sophisticated type inference. Maybe instead of returning `bool`, our `is_metaclass` function here should return an enum with three variants: `Yes`, `No` and `Maybe`. For classes with dynamic bases such as this class and the class reported in #12736, the function would return `Maybe`, and we would not emit either N804 or N805 on the methods in the class.

**Comment by AlexWaygood** [comment]
> Maybe instead of returning `bool`, our `is_metaclass` function here should return an enum with three variants: `Yes`, `No` and `Maybe`. For classes with dynamic bases such as this class and the class reported in [#12736](https://github.com/astral-sh/ruff/issues/12736), the function would return `Maybe`, and we would not emit either N804 or N805 on the methods in the class.

If anybody wants to help out with this issue: let's try doing something like this
